From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 16 Jun 2019 13:00:34 +0100
Subject: tcp: Avoid ABI change for DoS fixes
Forwarded: not-needed

"tcp: tcp_fragment() should apply sane memory limits" adds a new Linux
MIB counter.  This adds another element to the array in struct
linux_mib.  Since that is always allocated by built-in code, it's a
backward-compatible change and we can hide the added element from
genksyms.

Also hide the added element from modules, as they won't be able to
rely on its being present until we bump ABI.

---
--- a/include/uapi/linux/snmp.h
+++ b/include/uapi/linux/snmp.h
@@ -265,7 +265,9 @@ enum
 	LINUX_MIB_TCPWANTZEROWINDOWADV,		/* TCPWantZeroWindowAdv */
 	LINUX_MIB_TCPSYNRETRANS,		/* TCPSynRetrans */
 	LINUX_MIB_TCPORIGDATASENT,		/* TCPOrigDataSent */
+#if !defined(__KERNEL__) || (!defined(__GENKSYMS__) && !defined(MODULE))
 	LINUX_MIB_TCPWQUEUETOOBIG,		/* TCPWqueueTooBig */
+#endif
 	__LINUX_MIB_MAX
 };
 
